docs/pages/workflow/continuous-native-generation.mdx

```diff
@@ -25,91 +26,254 @@ Expo as a framework enables **Continuous Native Generation** by combining the fo
+
+For example, some config plugins aren't idempotent. When a project utilizes multiple "dangerous modifiers" to add regex changes to an app's code, it can lead to unexpected behavior. This is why using the `--clean` option is the safest way to use the prebuild command and is generally recommended in most cases.
+
+#### Using `--clean` option
+
+When using the `--clean` option, you'll be warned to have a clean git status due its destructive nature. This prompt is optional and will be skipped when encountered in CI. You can disable this check by enabling the environment variable `EXPO_NO_GIT_STATUS=1`.
+
+This prompts aims to encourage developers to add **android** and **ios** directories to the project's **.gitignore** or **.easignore**. This ensures that these directories are always generated by prebuild. However, this can make custom config plugins harder to develop so we haven't introduced any mechanism to enforce this behavior.
+
+There are cases where developers may want to swap between workflows often. For example, you may want to build custom functionality natively in Android Studio and Xcode, and then move that functionality into local config plugins.
 
-While library authors' adoption of CNG is not a prerequisite for using `npx expo prebuild`, it significantly enhances the user experience. If a library author has not yet adopted CNG, users can still use `npx expo prebuild` by creating local Config Plugins to modify the native generation pipeline. This flexibility ensures that the adoption of CNG is accessible and beneficial to all users within the React Native community.
+## Templates
 
-## Success
+You can customize how the native folders are generated by [config plugins](/config-plugins/introduction). Many config plugins already exist for lots of modifications, and community libraries often ship their own as well. You can [see a list of some popular plugins](https://github.com/expo/config-plugins) for more information.
 
-CNG is designed to empower developers to build better apps than they normally could, by dividing configuration into testable and maintainable pieces. This allows developers to build more powerful apps without compromising on iteration speed.
+Prebuild starts from template files which are then modified with config plugins. The template files are based on the Expo SDK version and come from the npm package [`expo-template-bare-minimum`](https://github.com/expo/expo/tree/main/templates/expo-template-bare-minimum). You can change the template used by passing `--template /path/to/template.tgz` to the `npx expo prebuild` command. This is not generally recommended because the base modifiers in `@expo/prebuild-config` make some undocumented assumptions about the template files, so it may be tricky to maintain your custom template.
```

Comment:
```suggestion
Prebuild starts from template files, which are then modified with config plugins. The template files are based on the Expo SDK version and come from the npm package [`expo-template-bare-minimum`](https://github.com/expo/expo/tree/main/templates/expo-template-bare-minimum). You can change the template used by passing `--template /path/to/template.tgz` to the `npx expo prebuild` command. This is not generally recommended because the base modifiers in `@expo/prebuild-config` make some undocumented assumptions about the template files, so it may be tricky to maintain your custom template.
```

Reply:
I am not sure why a comma is required here to break the sentence.

Reply:
No strong opinion, but I think it's a restrictive clause. Intuitively while speaking, you would pause before the "which", which is often (but not always) suggestive of a comma.